packages/neptune-extension/src/common/components/button/Button.tsx

```diff
@@ -0,0 +1,61 @@
+// Libs
+import React, { ButtonHTMLAttributes } from 'react';
+import { bemBlock } from 'common/utils/bem';
+
+import { getEventHandler, EventHandler } from 'common/utils/events';
+
+// Module
+import './Button.less';
+
+const block = bemBlock('n-button');
+
+export interface ButtonProps extends ButtonHTMLAttributes<HTMLButtonElement>{
+  className?: string
+  disabled?: boolean
+  pressed?: boolean
+  size?: 'medium' | 'large'
+  type?: 'button' | 'reset' | 'submit'
+  variant?: 'primary' | 'secondary'
+  component?: string | React.FC<any> | React.ComponentClass<any>
+  onClick?: EventHandler
+};
+
+const Button: React.FC<ButtonProps> = ({
+  children,
+  className,
+  disabled = false,
+  pressed = false,
+  size = 'medium',
+  type = 'button',
+  variant = 'primary',
+  component = 'button',
+  onClick,
+  ...props
+}) => {
+
+  const cssClasses = block({
+    modifiers: {
+      size,
+      variant,
+      disabled,
+      pressed,
+    },
+    extra: [className],
+  });
+
+  const onClickHandler = getEventHandler(onClick);
```

Comment:
Maybe I miss sth. obvious but what is wrong with a plain onClick, so that we need custom handlers ?

Reply:
You can avoid passing arrow functions to Button, but use custom object construction:
```
const onClick = {
    payload: { someId, someData },
    handler: (ev, {payload}) => dispatch(Action({someId, someData}));
}
```